corehq/apps/cowin/repeaters.py

```diff
@@ -0,0 +1,73 @@
+from django.utils.translation import ugettext_lazy as _
+
+from corehq import toggles
+from corehq.apps.cowin.repeater_generators import (
+    BeneficiaryRegistrationPayloadGenerator,
+    BeneficiaryVaccinationPayloadGenerator,
+)
+from corehq.apps.hqcase.utils import update_case
+from corehq.motech.repeaters.models import CaseRepeater
+
+
+class BaseCOWINRepeater(CaseRepeater):
+    class Meta:
+        app_label = 'repeaters'
+
+    @classmethod
+    def available_for_domain(cls, domain):
+        return toggles.COWIN_INTEGRATION.enabled(domain)
+
+    def get_headers(self, repeat_record):
+        headers = super(BaseCOWINRepeater, self).get_headers(repeat_record)
+        headers.update({
+            'Accept-Language': 'en_US',
+            'User-Agent': '',
+            'Accept': 'application/json',
+            'Content-Type': 'application/json',
+        })
+        return headers
+
+
+class BeneficiaryRegistrationRepeater(BaseCOWINRepeater):
+    payload_generator_classes = (BeneficiaryRegistrationPayloadGenerator,)
+    friendly_name = _("Register beneficiaries on COWIN")
+
+    def get_headers(self, repeat_record):
+        headers = super(BeneficiaryRegistrationRepeater, self).get_headers(repeat_record)
+        headers.update({
+            'X-Api-Key': self.connection_settings.plaintext_password,
+        })
+        return headers
+
+    def handle_response(self, response, repeat_record):
+        attempt = super().handle_response(response, repeat_record)
+        if response.status_code == 200:
+            beneficiary_reference_id = response.json()['beneficiary_reference_id']
```

Comment:
What do we expect `response.json()` to look like? It might be nice to document that here, or (maybe better) mocked in a unit test.

Reply:
https://github.com/dimagi/commcare-hq/pull/29949/commits/5de6bc3b512fbf97205907ed996fa382305ae146